filters/question_word/filter.py

```diff
@@ -0,0 +1,217 @@
+        self.keywords = keywords
+        self.nlp = spacy.load("en_core_web_sm")
+
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        tokenized = self.nlp(question, disable=["parser", "tagger", "ner"])
+        tokenized = [token.text.lower() for token in tokenized]
+        contained_keywords = set(tokenized).intersection(set(self.keywords))
+        return bool(contained_keywords)
+
+
+class WhereQuestion(BaseFilter):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self, keywords=["where"]):
+        super().__init__(keywords=keywords)
+
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        return super().filter(context,question,answers)
+
+
+class WhatQuestion(BaseFilter):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self, keywords=["what"]):
+        super().__init__(keywords=keywords)
+
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        return super().filter(context,question,answers)
+
+
+class WhoQuestion(BaseFilter):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self, keywords=["who"]):
+        super().__init__(keywords=keywords)
+
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        return super().filter(context,question,answers)
+
+class WhichQuestion(BaseFilter):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self, keywords=["which"]):
+        super().__init__(keywords=keywords)
+
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        return super().filter(context,question,answers)
+
+
+class WhyQuestion(BaseFilter):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self, keywords=["why"]):
+        super().__init__(keywords=keywords)
+
+    def filter(self,context:str = None ,question: str = None, answers: List[str]= None) -> bool:
+        return super().filter(context,question,answers)
+
+
+class WhenQuestion(BaseFilter):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self, keywords=["when"]):
+        super().__init__(keywords=keywords)
+
+    def filter(self,context:str = None ,question: str = None, answers:List[str] =  None) -> bool:
+        return super().filter(context,question,answers)
+
+
+class NumericQuestion(QuestionAnswerOperation):
+    tasks = [TaskType.QUESTION_GENERATION, TaskType.QUESTION_ANSWERING]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+        self.nlp = spacy.load("en_core_web_sm")
```

Comment:
Also use the following line to load spacy else all your filters would accumulate memory: 
`self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")`

Corresponding import:
`from initialize import spacy_nlp`

Reply:
Thanks for the review @kaustubhdhole 
I have made the suggested changes and regarding the UIUC taxonomy, indeed will give that a try too